File: clients/android/NewsBlur/src/com/newsblur/activity/SocialFeedReading.java
package com.newsblur.activity;

import android.os.Bundle;

import com.newsblur.domain.SocialFeed;
import com.newsblur.util.FeedUtils;
import com.newsblur.util.UIUtils;

public class SocialFeedReading extends Reading {

    @Override
    protected void onCreate(Bundle savedInstanceBundle) {
        super.onCreate(savedInstanceBundle);
        SocialFeed socialFeed = dbHelper.getSocialFeed(fs.getSingleSocialFeed().getKey());
        if (socialFeed == null) finish(); // don't open fatally stale intents
        UIUtils.setupToolbar(this, socialFeed.photoUrl, socialFeed.feedTitle, iconLoader, false);
    }

}

File: clients/android/NewsBlur/app/src/main/java/com/newsblur/activity/SocialFeedReading.kt
package com.newsblur.activity

import android.os.Bundle
import androidx.lifecycle.lifecycleScope
import com.newsblur.util.UIUtils
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

class SocialFeedReading : Reading() {

    override fun onCreate(savedInstanceBundle: Bundle?) {
        super.onCreate(savedInstanceBundle)

        lifecycleScope.launch(Dispatchers.IO) {
            val socialFeed = dbHelper.getSocialFeed(fs!!.singleSocialFeed.key)
            withContext(Dispatchers.Main) {
                if (socialFeed != null) {
                    UIUtils.setupToolbar(this@SocialFeedReading, socialFeed.photoUrl, socialFeed.feedTitle, iconLoader, false)
                } else {
                    finish()
                }
            }
        }
    }
}
